Userland/Libraries/LibUnicode/Normalize.cpp

```diff
@@ -122,12 +122,18 @@ static u32 combine_code_points(u32 a, u32 b)
 {
     Array<u32, 2> const points { a, b };
     // FIXME: Do something better than linear search to find reverse mappings.
-    for (auto const& mapping : Unicode::code_point_decompositions()) {
+    size_t index = { 0 };
+    for (;;) {
```

Comment:
Are we going to end up in an infinite loop if we enter this block below?
```c++
        if (mapping.tag == CompatibilityFormattingTag::Canonical && mapping.decomposition == points) {
            if (code_point_has_property(mapping.code_point, Property::Full_Composition_Exclusion))
                continue;
```

`index` won't get updated so we'll just keep repeating the same lookup?

Reply:
Yes, yes we are. Ouch.

Reply:
Should be fixed. :)